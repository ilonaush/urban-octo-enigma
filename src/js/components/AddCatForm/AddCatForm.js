import React, {Component} from 'react';
import "./AddCatForm.styl";
import Input from "../Input/Input";
import PropTypes from 'prop-types';

export default class AddCatForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            cat: {
                name: '',
                age: '',
                color: '',
                reason: '',
                location: '',
                health: ''
            }
        };
    }

    /**
     * handles input change
     * @param value
     * @param name
     */
    handleChange = ({target: {value, name}}) => {
        this.setState((prevState) => ({
            cat: {
                ...prevState.cat,
                [name]: value
            }
        }))
    };


    render() {
        const {cat} = this.state;
        const {handleSubmit} = this.props;
        return (
            <form className='addWorkerForm' onSubmit={(e) => handleSubmit(e, cat)}>
                <h3>Please share some info about the cat with us</h3>
                <Input id='name-input'
                       name='name'
                       placeholder='Name'
                       onChange={this.handleChange}
                       value={cat.name}
                       required
                        />
                <Input name='age'
                       placeholder='Age'
                       onChange={this.handleChange}
                       value={cat.age}
                       required
                />
                <Input name='color'
                       placeholder='Color'
                       onChange={this.handleChange}
                       value={cat.color}
                       required
                />
                <textarea name='reason'
                       placeholder='Reason'
                       onChange={this.handleChange}
                       value={cat.reason}
                />
                <Input name='location'
                       placeholder='Location'
                       onChange={this.handleChange}
                       value={cat.location}
                       required
                />
                <select
                    value={cat.health}
                    name='health'
                    onChange={this.handleChange}>
                        <option value="Satisfying">Satisfying</option>
                        <option value="Minor problems">Minor problems</option>
                        <option value="Not satisfying">Not satisfying</option>
                </select>
                <button id='submit-btn' type='submit'>Accept the cat</button>
            </form>
        );
    }
}

AddCatForm.propTypes = {
    handleSubmit: PropTypes.func,
};

