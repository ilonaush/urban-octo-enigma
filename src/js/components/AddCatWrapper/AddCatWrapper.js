import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import actions from "../../reducers/actions";
import {connect} from "react-redux";
import AddCatForm from "../AddCatForm/AddCatForm";
import "./AddCatWrapper.styl";

export class AddCatWrapper extends Component {

    /**
     * handles submit on add WorkerForm, dispatches an action
     * @param e
     * @param cat
     */
    handleSubmit = (e, cat) => {
        e.preventDefault();
        const catInfo = {
            name: cat.name,
            id: Date.now(),
            age: cat.age,
            color: cat.color,
            reason: cat.reason,
            location: cat.location,
            health: cat.health
        };
        this.props.addCat(catInfo).then(() => this.props.history.push('/'))
    };

    render() {
        return (
            <Fragment>
                <AddCatForm handleSubmit={this.handleSubmit}/>
                <img  className='cat-bg' src="/images/cat-bg-2.png" alt=""/>
            </Fragment>
        );
    }
}

AddCatWrapper.propTypes = {
    cats: PropTypes.array,
    addCat: PropTypes.func
};

export default connect(
    (state) => ({cats: state.cats}),
    (dispatch) => ({addCat: (cat) => dispatch(actions.addCat(cat))})
)(AddCatWrapper);
