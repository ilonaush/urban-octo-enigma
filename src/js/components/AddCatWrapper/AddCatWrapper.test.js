import React from 'react';
import {AddCatWrapper} from './AddCatWrapper';
import {mount} from "enzyme/build";
import AddCatForm from "../AddCatForm/AddCatForm";


describe('AddCatWrapper', function () {

  beforeEach(() => {
    this.wrapper = mount(<AddCatWrapper/>);
  });

  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });

  it('calls submit function  when  the form is submitted', async () => {
    const historyMock = {push: jest.fn()};
    const mockAddCatAction = jest.fn().mockResolvedValue(43);
    const props = {
      addCat: mockAddCatAction,
      history: historyMock
    };

    this.wrapper.setProps(props);

    const AddWorkerFormComponent = this.wrapper.find(AddCatForm);
    expect(mockAddCatAction.mock.calls.length).toBe(0);
    AddWorkerFormComponent.simulate('submit');
    expect(mockAddCatAction.mock.calls.length).toBe(1)
  });
});

