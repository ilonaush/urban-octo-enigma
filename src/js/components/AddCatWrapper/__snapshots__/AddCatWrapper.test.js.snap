// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCatWrapper renders correctly 1`] = `
<AddCatWrapper>
  <AddCatForm
    handleSubmit={[Function]}
  >
    <form
      className="addWorkerForm"
      onSubmit={[Function]}
    >
      <h3>
        Please share some info about the cat with us
      </h3>
      <Input
        id="name-input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        value=""
      >
        <input
          id="name-input"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          value=""
        />
      </Input>
      <Input
        name="age"
        onChange={[Function]}
        placeholder="Age"
        required={true}
        value=""
      >
        <input
          name="age"
          onChange={[Function]}
          placeholder="Age"
          required={true}
          value=""
        />
      </Input>
      <Input
        name="color"
        onChange={[Function]}
        placeholder="Color"
        required={true}
        value=""
      >
        <input
          name="color"
          onChange={[Function]}
          placeholder="Color"
          required={true}
          value=""
        />
      </Input>
      <textarea
        name="reason"
        onChange={[Function]}
        placeholder="Reason"
        value=""
      />
      <Input
        name="location"
        onChange={[Function]}
        placeholder="Location"
        required={true}
        value=""
      >
        <input
          name="location"
          onChange={[Function]}
          placeholder="Location"
          required={true}
          value=""
        />
      </Input>
      <select
        name="health"
        onChange={[Function]}
        value=""
      >
        <option
          value="Satisfying"
        >
          Satisfying
        </option>
        <option
          value="Minor problems"
        >
          Minor problems
        </option>
        <option
          value="Not satisfying"
        >
          Not satisfying
        </option>
      </select>
      <button
        id="submit-btn"
        type="submit"
      >
        Accept the cat
      </button>
    </form>
  </AddCatForm>
  <img
    alt=""
    className="cat-bg"
    src="/images/backgrounds/cat-bg-2.png"
  />
</AddCatWrapper>
`;
